Unable to use libsrtp installed with other prefix than /usr

I am trying to compile janus-gateway 0.9.1 against libsrtp 2.3.0 on ubuntu bionic.

If I pass other prefix than /usr while compiling libsrtp it fails

```
checking for srtp_init in -lsrtp2... no
checking for srtp_init in -lsrtp... no
configure: error: libsrtp and libsrtp2 not found. See README.md for installation instructions
```

e.g. I do 

```
./configure --prefix=/opt/app/libsrtp --enable-openssl
make shared_library
sudo make install
```

then I pass /opt/libsrtp/lib/pkgconfig  to the PKG_CONFIG_PATH environment variable while configuring janus-gateway.

pkg-config properly resolves the library

```
root@2c76424c6db8:/build/janus-gateway# PKG_CONFIG_PATH=/opt/app/libnice/lib/pkgconfig:/opt/app/libsrtp/lib/pkgconfig pkg-config --cflags libsrtp2
-I/opt/app/libsrtp/include
root@2c76424c6db8:/build/janus-gateway# PKG_CONFIG_PATH=/opt/app/libnice/lib/pkgconfig:/opt/app/libsrtp/lib/pkgconfig pkg-config --libs libsrtp2
-L/opt/app/libsrtp/lib -lsrtp2 -lcrypto -ldl -pthread
```

but janus-gateway's configure scripts seem to fail in such case.

--- a/Makefile.am
+++ b/Makefile.am
@@ -144,6 +144,8 @@ janus_SOURCES = \
 janus_CFLAGS = \
 	$(AM_CFLAGS) \
 	$(JANUS_CFLAGS) \
+	$(LIBSRTP_CFLAGS) \
+	$(LIBCURL_CFLAGS) \
 	-DPLUGINDIR=\"$(plugindir)\" \
 	-DTRANSPORTDIR=\"$(transportdir)\" \
 	-DEVENTDIR=\"$(eventdir)\" \
@@ -156,6 +158,8 @@ janus_LDADD = \
 	$(BORINGSSL_LIBS) \
 	$(JANUS_LIBS) \
 	$(JANUS_MANUAL_LIBS) \
+	$(LIBSRTP_LDFLAGS) $(LIBSRTP_LIBS) \
+	$(LIBCURL_LDFLAGS) $(LIBCURL_LIBS) \
 	$(NULL)
 
 dist_man1_MANS = janus.1
@@ -317,9 +321,9 @@ events_libadd = \
 if ENABLE_SAMPLEEVH
 event_LTLIBRARIES += events/libjanus_sampleevh.la
 events_libjanus_sampleevh_la_SOURCES = events/janus_sampleevh.c
-events_libjanus_sampleevh_la_CFLAGS = $(events_cflags)
-events_libjanus_sampleevh_la_LDFLAGS = $(events_ldflags) -lcurl -lm
-events_libjanus_sampleevh_la_LIBADD = $(events_libadd)
+events_libjanus_sampleevh_la_CFLAGS = $(events_cflags) $(LIBCURL_CFLAGS)
+events_libjanus_sampleevh_la_LDFLAGS = $(events_ldflags) $(LIBCURL_LDFLAGS) $(LIBCURL_LIBS) -lm
+events_libjanus_sampleevh_la_LIBADD = $(events_libadd) $(LIBCURL_LIBADD)
 conf_DATA += conf/janus.eventhandler.sampleevh.jcfg.sample
 EXTRA_DIST += conf/janus.eventhandler.sampleevh.jcfg.sample
 endif
@@ -582,11 +586,13 @@ janus_pp_rec_SOURCES = \
 
 janus_pp_rec_CFLAGS = \
 	$(AM_CFLAGS) \
+	$(LIBCURL_CFLAGS) \
 	$(POST_PROCESSING_CFLAGS) \
 	$(NULL)
 
 janus_pp_rec_LDADD = \
 	$(POST_PROCESSING_LIBS) \
+	$(LIBCURL_LDFLAGS) $(LIBCURL_LIBS) \
 	$(NULL)
 
 BUILT_SOURCES += postprocessing/pp-cmdline.c postprocessing/pp-cmdline.h

--- a/configure.ac
+++ b/configure.ac
@@ -387,51 +387,29 @@ AC_CHECK_LIB([dl],
 
 AM_CONDITIONAL([ENABLE_LIBSRTP_2], false)
 AS_IF([test "x$enable_libsrtp2" != "xno"],
-      [AC_CHECK_LIB([srtp2],
-                    [srtp_init],
-                    [
-                      AC_CHECK_HEADER([srtp2/srtp.h],
-                                      [
-                                        AC_DEFINE(HAVE_SRTP_2)
-                                        JANUS_MANUAL_LIBS="${JANUS_MANUAL_LIBS} -lsrtp2"
-                                        enable_libsrtp2=yes
-                                        AM_CONDITIONAL([ENABLE_LIBSRTP_2], true)
-                                        AC_CHECK_LIB([srtp2],
-                                                     [srtp_crypto_policy_set_aes_gcm_256_16_auth],
-                                                     [AC_DEFINE(HAVE_SRTP_AESGCM)],
-                                                     [AC_MSG_NOTICE([libsrtp2 does not have support for AES-GCM profiles])])
-                                      ],
-                                      [
-                                        AS_IF([test "x$enable_libsrtp2" = "xyes"],
-                                              [AC_MSG_ERROR([libsrtp2 headers not found. See README.md for installation instructions or use --disable-libsrtp-2 to try and autodetect libsrtp 1.5.x instead])])
-                                      ])
-                    ],
-                    [
-                      AS_IF([test "x$enable_libsrtp2" = "xyes"],
-                            [AC_MSG_ERROR([libsrtp2 not found. See README.md for installation instructions or use --disable-libsrtp-2 to try and autodetect libsrtp 1.5.x instead])])
-                    ])
+      [PKG_CHECK_MODULES([LIBSRTP],
+                         [libsrtp2],
+                         [
+                          AC_DEFINE(HAVE_SRTP_2)
+                          AC_DEFINE(HAVE_SRTP_AESGCM)
+                          enable_libsrtp2=yes
+                          AM_CONDITIONAL([ENABLE_LIBSRTP_2], true)
+                         ],
+                         [
+                          AS_IF([test "x$enable_libsrtp2" = "xyes"],
+                                [AC_MSG_ERROR([libsrtp2 headers not found. See README.md for installation instructions or use --disable-libsrtp-2 to try and autodetect libsrtp 1.5.x instead])])
+                         ])
       ])
 AM_COND_IF([ENABLE_LIBSRTP_2],
            [],
-           [AC_CHECK_LIB([srtp],
-                         [srtp_init],
-                         [
-                           AC_CHECK_HEADER([srtp/srtp.h],
-                                           [
-                                             JANUS_MANUAL_LIBS="${JANUS_MANUAL_LIBS} -lsrtp"
-                                             enable_libsrtp2=no
-                                             PKG_CHECK_MODULES([SRTP15X],
-                                                               [
-                                                                 libsrtp >= 1.5.0
-                                                               ])
-                                             AC_CHECK_LIB([srtp],
-                                                          [crypto_policy_set_aes_gcm_256_16_auth],
-                                                          [AC_DEFINE(HAVE_SRTP_AESGCM)],
-                                                          [AC_MSG_NOTICE([libsrtp does not have support for AES-GCM profiles])])
-                                           ],
-                                           [AC_MSG_ERROR([libsrtp and libsrtp2 headers not found. See README.md for installation instructions])])
-                         ],
-                         [AC_MSG_ERROR([libsrtp and libsrtp2 not found. See README.md for installation instructions])])
+           [PKG_CHECK_MODULES([LIBSRTP],
+                              [libsrtp >= 1.5.0],
+                              [
+                               AC_DEFINE(HAVE_SRTP_AESGCM)
+                               enable_libsrtp2=no
+                              ],
+                              [AC_MSG_ERROR([libsrtp and libsrtp2 not found. See README.md for installation instructions])
+                              ])
            ])
 
 AC_CHECK_LIB([usrsctp],
@@ -454,7 +432,6 @@ PKG_CHECK_MODULES([LIBCURL],
                   [libcurl],
                   [
                     AC_DEFINE(HAVE_LIBCURL)
-                    JANUS_MANUAL_LIBS="${JANUS_MANUAL_LIBS} -lcurl"
                     AS_IF(
                       [test "x$enable_turn_rest_api" != "xno"],
                       [

